File: code/C++/TicTacToe.cpp
#include "TicTacToeBoard.h"
#include <cstdlib>
#include <iostream>
#include <iterator>
#include <forward_list>
#include <string>
#include <random>
#include <vector>
#include <algorithm> 

void generateAllGames(TicTacToeBoard board)
{
	vector<string> moves=board.legalMoves();
	for (vector<string>::iterator it = moves.begin(); it != moves.end(); ++it){
		board.makeMove(*it);
		if (board.isGameOver()) {
			for (auto const& move: board.playedMoves()) cout << move; 
			cout << endl;
		} else {
			generateAllGames(TicTacToeBoard(board));
		}
		board.undoLastMove();
	}
}

int main() 
{
	generateAllGames(TicTacToeBoard());

	return(EXIT_SUCCESS);
}

File: code/Java/TicTacToe.java
import java.util.ArrayList;

public class TicTacToe {
	public void generateAllGames(TicTacToeBoard board) {
		ArrayList<String> moves=board.legalMoves();
		for (String move : moves) {
			board.makeMove(move);
			if (board.isGameOver()) {
				for (String amove : board.playedMoves()) {
					System.out.print(amove);
				}
				System.out.println();
			} else {
				generateAllGames(new TicTacToeBoard(board));
			}
			board.undoLastMove();
		}
	}

	public static void main(String[] args) {
		TicTacToe t=new TicTacToe();
		TicTacToeBoard b=new TicTacToeBoard();
		t.generateAllGames(b);
	}
}
